app/controllers/practices_controller.rb

```diff
@@ -34,19 +36,20 @@ def update
 end
 
 private
-def format_dates_by_term(month)
+
+def format(progresses, months)
   end_date = Time.zone.now
-  start_date = end_date.months_ago(month)
+  start_date = end_date.months_ago(months)
   date_per_week = []
 
   while (start_date <= end_date)
-    if date_per_week.empty? || date_per_week.last.last.wday == 6
-      date_per_week << [start_date]
+    if date_per_week.empty? || date_per_week.last.any? {|k, _| k.wday == 6}
+      date_per_week << \
+          {start_date => progresses.select{|p| p.done_at == start_date.to_date}.last}
     else
-      date_per_week.last << start_date
+      date_per_week.last[start_date] = progresses.select{|p| p.done_at == start_date.to_date}.last
```

Comment:
__[reek]__
* [DuplicateMethodCall] calls date_per_week.last 2 times
* [DuplicateMethodCall] calls p.done_at 2 times
* [DuplicateMethodCall] calls p.done_at == start_date.to_date 2 times
* [DuplicateMethodCall] calls progresses.select 2 times
* [DuplicateMethodCall] calls progresses.select do |p| ... end 2 times
* [DuplicateMethodCall] calls progresses.select do |p| ... end.last 2 times
* [DuplicateMethodCall] calls start_date.to_date 2 times
* [UncommunicativeVariableName] has the variable name 'p'